drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -0,0 +1,918 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT nxp_ehci
+
+#include <soc.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/drivers/usb/udc.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#include "udc_common.h"
+#include "usb.h"
+#include "usb_device_config.h"
+#include "usb_device_mcux_drv_port.h"
+#include "usb_device_ehci.h"
+#ifdef CONFIG_DT_HAS_NXP_USBPHY_ENABLED
```

Comment:
Please remove `#ifdef CONFIG_DT_HAS_NXP_USBPHY_ENABLED`, it should not be necessary.

Reply:
As I explained, NXP has may platforms. Even the same USB controller IP is used, the PHY is used or not in different platforms. For example: (1) LPC54608 uses the IP3511 driver, the phy is not used; LPC55S69 uses the IP3511 driver, the phy is used. (2) RW610 uses the EHCI driver, the phy is not used; RTxxxx series uses the EHCI driver, the phy is used.
So, the MACRO is needed here.

Reply:
I do not care how many platforms NXP has. There is no need to include this header conditionally.

Reply:
why does it matter?